kpatch-build/lookup.c

```diff
@@ -48,12 +48,77 @@ struct lookup_table {
 	int fd, nr;
 	Elf *elf;
 	struct symbol *syms;
+	struct symbol *local_syms;
 };
 
 #define for_each_symbol(ndx, iter, table) \
 	for (ndx = 0, iter = table->syms; ndx < table->nr; ndx++, iter++)
 
-struct lookup_table *lookup_open(char *path)
+static void resolve_dup_file_symbol(struct lookup_table *table, char *hint,
+				    struct sym_compare_type *locals)
+{
+	struct symbol *sym, *prev;
+	int i, in_file = 0;
+	int multi_base = 0;
+	struct sym_compare_type *local_index;
+
+	for_each_symbol(i, sym, table) {
+		if (sym->type == STT_FILE) {
+			if (in_file) {
+				if (table->local_syms) {
+					multi_base = 1;
```

Comment:
I found the multi_base variable confusing.  It might be clearer to either rename it to something like 'found_dup'.  Or, even better, get rid of it and instead use a goto to the error path, as that would make the code flow clearer IMO.